WcaOnRails/app/models/person.rb

```diff
@@ -1,13 +1,99 @@
 class Person < ActiveRecord::Base
-  self.table_name = "Persons"
+  self.table_name = "rails_persons"
   self.primary_key = "id"
-  has_one :user, foreign_key: "wca_id"
-  has_many :results, foreign_key: "personId"
+
+  has_one :user, primary_key: "wca_id", foreign_key: "wca_id"
+  has_many :results, primary_key: "wca_id", foreign_key: "personId"
   has_many :competitions, -> { distinct }, through: :results
-  has_many :ranksAverage, foreign_key: "personId", class_name: "RanksAverage"
-  has_many :ranksSingle, foreign_key: "personId", class_name: "RanksSingle"
+  has_many :ranksAverage, primary_key: "wca_id", foreign_key: "personId", class_name: "RanksAverage"
+  has_many :ranksSingle, primary_key: "wca_id", foreign_key: "personId", class_name: "RanksSingle"
+
+  scope :current, -> { where(subId: 1) }
+
+  validates :name, presence: true
+  validates :countryId, presence: true
+
+  before_validation :unpack_dates
+  private def unpack_dates
```

Comment:
1. I've stared at this function for a while now, and I just can't figure out what is what between `@dob`, the `attr_writer :dob` and the function `dob`. This needs to be made readable.

2. This function mixes validation with whatever the other thing going on is. It would probably be better to have the validation part be a normal validation.